drivers/adc/ad7616/iio_ad7616.c

```diff
@@ -0,0 +1,383 @@
+	OSR_AVAILABLE_ID,
+};
+
+/**
+ * @brief	Check if the channel number correspond to the channel A of AD7616
+ * @param	channel_id - Id of the channel
+ * @returns	1 if true, 0 otherwise
+ */
+static int32_t is_channel_a(uint32_t channel_id)
+{
+	return channel_id < CHANNEL_SOURCES_NUMBER;
+}
+
+/**
+ * @brief       Read the debug register value
+ * @param       dev[in, out]- Pointer to IIO device instance
+ * @param       reg[in]- Register address to read from
+ * @param       readval[out]- Pointer to variable to read data into
+ * @return      0 in case of success, negative value otherwise
+ */
+static int32_t iio_ad7616_debug_reg_read(void *device, uint32_t reg,
+		uint32_t *val)
+{
+	struct ad7616_iio_dev *iio_dev = device;
+	struct ad7616_dev *dev = iio_dev->ad7616_dev;
+
+	return ad7616_read(dev,  (uint8_t) reg, (uint16_t *)val);
+}
+
+/**
+ * @brief       Write the debug register value
+ * @param       dev[in, out]- Pointer to IIO device instance
+ * @param       reg[in]- Register address to read from
+ * @param       writeval[out]- Pointer to variable to write data into
+ * @return      0 in case of success, negative value otherwise
+ */
+static int32_t iio_ad7616_debug_reg_write(void *device, uint32_t reg,
+		uint32_t val)
+{
+	struct ad7616_iio_dev *iio_dev = device;
+	struct ad7616_dev *dev = iio_dev->ad7616_dev;
+
+	return ad7616_write(dev, (uint8_t)reg, (uint16_t)val);
+}
+
+static int get_raw(void *device,
+		   char *buf,
+		   uint32_t len,
+		   const struct iio_ch_info *channel,
+		   intptr_t id)
+{
+	int ret;
+	uint8_t index = is_channel_a(channel->ch_num) ? channel->ch_num :
+			channel->ch_num - CHANNEL_SOURCES_NUMBER;
+	struct ad7616_conversion_result res[RESULTS_NUMBER];
+	struct ad7616_iio_dev *iio_dev = device;
+	struct ad7616_dev *dev = iio_dev->ad7616_dev;
+
+	ret = ad7616_read_data_serial(dev, res, 1);
+	if (ret)
+		return ret;
+
+	if (is_channel_a(channel->ch_num))
+		return sprintf(buf, "%i", res[index].channel_a);
+
+	return sprintf(buf, "%i", res[index].channel_b);
+}
+
+static int get_scale(void *device,
+		     char *buf,
+		     uint32_t len,
+		     const struct iio_ch_info *channel,
+		     intptr_t id)
+{
+	struct ad7616_iio_dev *iio_dev = device;
+	struct ad7616_dev *dev = iio_dev->ad7616_dev;
+
+	enum ad7616_range range = is_channel_a(channel->ch_num) ?
+				  dev->va[channel->ch_num] : dev->vb[channel->ch_num - CHANNEL_SOURCES_NUMBER];
+
+	switch (range) {
+	case AD7616_10V:
+		return sprintf(buf, "10");
+	case AD7616_5V:
+		return sprintf(buf, "5");
+	case AD7616_2V5:
+		return sprintf(buf, "2.5");
+	default:
+		return -EINVAL;
+	}
+}
+
+static int set_over_sampling_ratio(struct ad7616_dev *dev,
+				   char *buf,
+				   uint32_t len)
+{
+	enum ad7616_osr osr;
+
+	sscanf(buf, "%u", (int *)&osr);
+	osr = log2(osr);
+
+	switch (osr) {
+	case AD7616_OSR_0:
+	case AD7616_OSR_2:
+	case AD7616_OSR_4:
+	case AD7616_OSR_8:
+	case AD7616_OSR_16:
+	case AD7616_OSR_32:
+	case AD7616_OSR_64:
+	case AD7616_OSR_128:
```

Comment:
can use the following syntax instead, to remove extra lines of code:
case AD7616_OSR_0 ... AD7616_OSR_128: